services/mailer.js

```diff
@@ -54,102 +54,108 @@ templates.render = (name, format = 'txt', context) => new Promise((resolve, reje
 
     return resolve(view(context));
   });
-}); // ends templates.render
+});
+
+const mailer = {};
+
+// enabled is true when the required configuration is available. When testing
+// is enabled, we will be simulating that emails are being sent, because in a
+// production system, emails should and would be sent.
+mailer.enabled = Boolean(
+  SMTP_HOST && SMTP_HOST.length > 0 &&
+  SMTP_USERNAME && SMTP_USERNAME.length > 0 &&
+  SMTP_PORT && SMTP_PORT.length > 0 &&
+  SMTP_PASSWORD && SMTP_PASSWORD.length > 0 &&
+  SMTP_FROM_ADDRESS && SMTP_FROM_ADDRESS.length > 0
+) || process.env.NODE_ENV === 'test';
+
+if (mailer.enabled) {
+  const options = {
+    host: SMTP_HOST,
+    auth: {
+      user: SMTP_USERNAME,
+      pass: SMTP_PASSWORD
+    }
+  };
 
-const options = {
-  host: SMTP_HOST,
-  auth: {
-    user: SMTP_USERNAME,
-    pass: SMTP_PASSWORD
+  if (SMTP_PORT) {
+    try {
+      options.port = parseInt(SMTP_PORT);
+    } catch (e) {
+      throw new Error('TALK_SMTP_PORT is not an integer');
+    }
+  } else {
+    options.port = 25;
   }
-};
 
-if (SMTP_PORT) {
-  try {
-    options.port = parseInt(SMTP_PORT);
-  } catch (e) {
-    throw new Error('TALK_SMTP_PORT is not an integer');
-  }
-} else {
-  options.port = 25;
+  mailer.transport = nodemailer.createTransport(options);
 }
 
-const defaultTransporter = nodemailer.createTransport(options);
-
-const mailer = module.exports = {
-
-  /**
-   * Create the new Task kue.
-   */
-  task: new kue.Task({
-    name: 'mailer'
-  }),
-
-  sendSimple({template, locals, to, subject}) {
-
-    if (!to) {
-      return Promise.reject('sendSimple requires a comma-separated list of "to" addresses');
-    }
+/**
+ * Create the new Task kue.
+ */
+mailer.task = new kue.Task({
+  name: 'mailer'
+});
+
+/**
+ * send will create a new message and send it.
+ */
+mailer.send = async ({template, locals, to, subject}) => {
+  if (!mailer.enabled) {
+    throw new Error('email is not enabled because required configuration is not available');
+  }
 
-    if (!subject) {
-      return Promise.reject('sendSimple requires a subject for the email');
+  // Attach the template locals.
+  attachStaticLocals(locals);
```

Comment:
Instead of modifying that was passed as an argument I'd recommend copying the locals e.g.  `{...locals, t: i18n.t, staticLocals}`. It's best to reduce side effects and keep functions pure.

Reply:
I can look into using lodash.merge, we don't have object spread on the server :(

Reply:
This is the same as object spread: `Object.assign({}, locals, {t: i18n.t}, staticLocals)`